egs/librispeech/ASR/transducer_stateless/decoder.py

```diff
@@ -82,17 +87,36 @@ def forward(self, y: torch.Tensor, need_pad: bool = True) -> torch.Tensor:
         Returns:
           Return a tensor of shape (N, U, embedding_dim).
         """
-        embeding_out = self.embedding(y)
+        embedding_out = self.embedding(y)
         if self.context_size > 1:
-            embeding_out = embeding_out.permute(0, 2, 1)
+            embedding_out = embedding_out.permute(0, 2, 1)
             if need_pad is True:
-                embeding_out = F.pad(
-                    embeding_out, pad=(self.context_size - 1, 0)
-                )
+                # Regarding the left or right context we are using,
+                # if we feed sequence [sos, a, b, c, d] to this decoder, and
+                # want to predict the sequence [a, b, c, d]. After padding to
+                # the left with context_size==2, the fed in sequence changes to
+                # [pad, sos, a, b, c, d], and we use `pad,sos` to predict `a`,
+                # `sos,a` to predict `b` ..., that is left context.
+                # if we feed sequence [b, c, d, blk, blk] to this decoder,
```

Comment:
Clarify before the "if" that you are now talking about the backwards version of the decoder.
E.g.
`For the "backwards" RNN-T: `